src/inverters/dto/custom-entity.ts

```diff
@@ -53,8 +53,8 @@ export function convertToCustomEntity(entity: any): CustomEntity {
         ...entity,
         toNum: (decimals?: number, invert?: boolean) => Utils.toNum(entity?.state, decimals, invert),
         isValid: () => entity?.state !== null && entity.state !== undefined && entity.state !== 'unknown' || false,
-        notEmpty: () => entity?.state !== '' || false,
```

Comment:
I dont think you should change this. ?? checks for a nullish value but we are looking for an empty string

Reply:
this might be the same as your other comment. can never be undefined after the `!==` evaluation, if `entity?.state` is undefined it will also be false.

Reply:
@Pho3niX90  so we need to cover all possible "invalid" states as edited above?